### PR TITLE
Update starknet-devnet to v0.3.1

### LINKED ISSUE #834: Update Starknet Devnet to v0.3.1

### PR DESCRIPTION
**Pull request by mkaput**
fix #834

### CODE REVIEW

**Review by kasperski95 — APPROVED** [review]

File: protostar/starknet_gateway/gateway_facade.py [hunk 1]
```diff
@@ -265,6 +265,7 @@ async def invoke(
                 address=account_address,
                 client=self._gateway_client,
                 signer=signer,
+                supported_tx_version=1,
```
[the remark this answers is not in the document]
Comment (mkaput) [reply protostar/starknet_gateway/gateway_facade.py:268]: https://starknetpy.readthedocs.io/en/latest/migration_guide.html#new-transaction-version

**Comment by mkaput** [comment]
@kasperski95 @MaksymilianDemitraszek @Radinyn ready for review

File: tests/integration/migrator/compiled_account_contract_tx_v0.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# OpenZeppelin Account implementation v0.3.1 copied from Starknet Devnet v0.3.0
```
Comment (kasperski95) [inline tests/integration/migrator/compiled_account_contract_tx_v0.py:1]: I think this type of fixtures should be shared across tests. Consider moving it to `tests/data`.

File: tests/integration/migrator/migrator_invoke_cheatcode_test.py [hunk 3]
```diff
@@ -55,3 +62,48 @@ async def test_waiting_for_acceptance(
 """,
         account_address=acc_address,
     )
+
+
+async def test_account_with_tx_version_0(
+    devnet_gateway_url: str, migrate: MigrateFixture, monkeypatch
```
Comment (kasperski95) [inline tests/integration/migrator/migrator_invoke_cheatcode_test.py:68]:
```suggestion
    devnet_gateway_url: str, migrate: MigrateFixture, monkeypatch: pytest.MonkeyPatch
```

File: tests/integration/migrator/migrator_invoke_cheatcode_test.py [hunk 4]
```diff
@@ -55,3 +62,48 @@ async def test_waiting_for_acceptance(
 """,
         account_address=acc_address,
     )
+
+
+async def test_account_with_tx_version_0(
```
Comment (kasperski95) [inline tests/integration/migrator/migrator_invoke_cheatcode_test.py:67]: Is there somewhere a corresponding test for tx_version_1?

**Review by kasperski95 — APPROVED** [review]

**Comment by kasperski95** [comment]
Consider updating
- https://docs.swmansion.com/protostar/docs/tutorials/deploying/cli
- ConfigurationProfileCLI::profile arg description

File: tests/integration/migrator/compiled_account_contract_tx_v0.py (hunk 2 above)
Comment (mkaput) [reply tests/integration/migrator/compiled_account_contract_tx_v0.py:1]: I guarantee you that it's not going to be used anywhere else.

File: protostar/starknet_gateway/gateway_facade.py [hunk 5]
```diff
@@ -441,3 +455,53 @@ def validate_cairo_inputs(abi: AbiType, inputs: List[int]):
         to_python_transformer(abi, "constructor", "inputs")(inputs)
     except DataTransformerException as ex:
         raise InputValidationException(str(ex)) from ex
+
+
+class AccountTxVersionDetector:
```
[the remark this answers is not in the document]
Comment (mkaput) [reply protostar/starknet_gateway/gateway_facade.py:460]: Done.

File: tests/integration/migrator/migrator_invoke_cheatcode_test.py (hunk 3 above)
Comment (mkaput) [reply tests/integration/migrator/migrator_invoke_cheatcode_test.py:68]: Done.

**Comment by mkaput** [comment]
> Consider updating
> 
>     * https://docs.swmansion.com/protostar/docs/tutorials/deploying/cli
> 
>     * ConfigurationProfileCLI::profile arg description

@kasperski95 checked, only one link was outdated. Otherwise the information there is still correct.

**Comment by kasperski95** [comment]
Somebody on the Discord was confused by the chain-id. We could include chain-id in the devnet example configuration. This is a subject for another PR anyway.

**Pull request merged**